src/proxyp/Protocol.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_PROXYP_ROTOCOL_H
+#define SQUID_PROXYP_ROTOCOL_H
```

Comment:
Please rename this file to `Elements.h`. We use that naming approach for FTP, and adaptation protocols.